sci-libs/vtk/vtk-9.1.0.ebuild

```diff
@@ -0,0 +1,601 @@
+		)
+	fi
+
+	if use qt5; then
+		mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_GUISupportQt="WANT" )
+		if use mysql || use postgres; then
+			mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_GUISupportQtSQL="WANT" )
+		fi
+		if use rendering; then
+			mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_RenderingQt="WANT" )
+		fi
+		if use views; then
+			mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_ViewsQt="WANT" )
+		fi
+	fi
+
+	if use rendering || use web || use all-modules; then
+		# needs patched version
+		mycmakeargs+=( -DVTK_MODULE_USE_EXTERNAL_VTK_libharu=OFF )
+	fi
+
+	if use rendering; then
+		mycmakeargs+=(
+			-DVTK_MODULE_ENABLE_VTK_IOExportGL2PS="WANT"
+			-DVTK_USE_SDL2=ON
+# FIXME: this should be default with VTK_USE_EXTERNAL=ON. Verify this!
+#			-DVTK_MODULE_USE_EXTERNAL_VTK_gl2ps=ON
+#			-DVTK_MODULE_USE_EXTERNAL_VTK_glew=ON
+#			-DVTK_MODULE_USE_EXTERNAL_VTK_libproj=ON
+		)
+	fi
+
+	# FIXME: needs improvements (and has been improved upstream by providing
+	# the VTK{,Large}DataFiles packages. Still need to thoroughly test
+	# if we need to download additional files.
+	if use test; then
+		ewarn "Testing requires VTK_FORBID_DOWNLOADS=OFF by upstream."
+		ewarn "Care has been taken to pre-download all required files."
+		ewarn "In case you find missing files, please inform me."
+		mycmakeargs+=(
+			-DVTK_BUILD_TESTING=ON
+			-DVTK_DATA_EXCLUDE_FROM_ALL=ON
+			-DVTK_FORBID_DOWNLOADS=OFF
+		)
+	else
+		mycmakeargs+=(
+			-DVTK_BUILD_TESTING=OFF
+			-DVTK_FORBID_DOWNLOADS=ON
+		)
+	fi
+
+	# TODO: remote the USE flag and enable by default
+	if use theora; then
+		mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_IOOggTheora="WANT" )
+	fi
+
+	# FIXME: upstream provides 4 threading models, as of 9.1.0. These are
+	# sequential, stdpthread, openmp and tbb. AFAICS all of them can be
+	# enabled at the same time. Sequential and Stdthread are enabled by
+	# default. The default selected type for the build is sequential.
+	# Assuming sequential < stdpthread < openmp < tbb wrt speed, although
+	# this is dependent on the actual scenario where threading is used.
+	if use tbb; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="TBB" )
+	elif use openmp; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="OpenMP" )
+	elif use threads; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="STDThread" )
+	else
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="Sequential" )
+	fi
+
+	if use openmp; then
+		mycmakeargs+=( -DVTK_SMP_ENABLE_OPENMP=ON )
+	fi
+
+	if use tbb; then
+		mycmakeargs+=( -DVTK_SMP_ENABLE_TBB=ON )
+	fi
+
+	# ON by default, hence the else clause is neede
+	if use threads; then
+		mycmakeargs+=( -DVTK_SMP_ENABLE_STDTHREAD=ON )
+	else
+		mycmakeargs+=( -DVTK_SMP_ENABLE_STDTHREAD=OFF )
+	fi
+
+	use java && export JAVA_HOME="${EPREFIX}/etc/java-config-2/current-system-vm"
+
+	if use mpi; then
+		export CC=mpicc
+		export CXX=mpicxx
+		export FC=mpif90
+		export F90=mpif90
+		export F77=mpif77
+	fi
+
+	cmake_src_configure
+}
+
+src_test() {
+	nonfatal virtx cmake_src_test
```

Comment:
Why nonfatal? Won't this prevent us from knowing if it fails?

Reply:
IIRC I added this in 9.0.1 in case virtx fails for some reason to avoid a false positive. According to the PMS nonfatal should return a non-zero exit status, in case the called function fails. So we could test for the return value to check if the tests failed.